mxcube3/ui/components/Login.jsx

```diff
@@ -1,50 +1,103 @@
+'use strict';
 import ReactDOM from 'react-dom';
-import React from 'react';
+import React, { Component, PropTypes } from 'react'
+import { Redirect, Router, Route } from 'react-router';
 import classNames from 'classnames';
 import "bootstrap-webpack!bootstrap-webpack/bootstrap.config.js";
-import { Nav, Input, ButtonInput } from "react-bootstrap";
+import { Input, ButtonInput, Form, Well } from "react-bootstrap";
+import './Login.css';
+import Main_login from './Main_login'
+import SampleViewContainer from '../containers/SampleViewContainer'
+import SampleGridContainer from '../containers/SampleGridContainer'
+import { Logging } from './Logging'
 
-export default class LoginForm extends React.Component {
+var LoginForm = React.createClass({
 
-    signIn() {
-        let proposal = this.refs.proposal.getValue();
-        let password = this.refs.password.getValue();
-        this.props.signIn(proposal, password);
-    }
+getInitialState: function() {
+    return {
+      login_info: {synchrotron_name: " ", beamline_name: " ", loginType: ""}
+    };
+  },
+  componentDidMount: function(){
+    $.ajax({
```

Comment:
Hi Matias,

Thanks for all the comments! I will fix them and make a new PR.

Best,
Jie

From: Matias Guijarro [mailto:notifications@github.com]
Sent: Monday, January 18, 2016 9:56 AM
To: mxcube/mxcube3
Cc: Jie Nan
Subject: Re: [mxcube3] A Separate Login Page (#96)


In mxcube3/ui/components/Login.jsx<https://github.com/mxcube/mxcube3/pull/96#discussion_r49973304>:

>

> -    signIn() {

> -        let proposal = this.refs.proposal.getValue();

> -        let password = this.refs.password.getValue();

> -        this.props.signIn(proposal, password);

> -    }

> +getInitialState: function() {

> +    return {

> +      login_info: {synchrotron_name: " ", beamline_name: " ", loginType: ""}

> +    };

> +  },

> +  componentDidMount: function(){

> +    $.ajax({

I am not sure but I think getting login info should go to the redux scheme: triggering an action, etc. instead of direct AJAX call ?
Instead of a query, another possibility to retrieve login info may be to have it set by login action through props.

—
Reply to this email directly or view it on GitHub<https://github.com/mxcube/mxcube3/pull/96/files#r49973304>.